src/storage/store/list_node_column.cpp

```diff
@@ -0,0 +1,160 @@
+#include "storage/store/list_node_column.h"
+
+#include "storage/copier/list_column_chunk.h"
+#include "storage/store/node_column.h"
+
+using namespace kuzu::common;
+
+namespace kuzu {
+namespace storage {
+
+void ListNodeColumn::scanInternal(transaction::Transaction* transaction,
```

Comment:
As discussed, currently in this PR we need to do two dynamic allocations for each scan (except for the one with node offset 0), which I assume will create performance bottlenecks for list scans. We definitely should profile and if necessary optimize further. Can you add a todo to keep a note of this?

A summary here:
1. If we switch to node-group per morsel later, we can simply keep a scan state that always keeps track of last offset from the last scan within the same node group.
2. For now, the alternative solution is to try to reuse `resultVector`, which avoids all dynamic allocation of offsetVectors. Keys are we use `resultVector` to scan base offset in storage first, and save the base offset as a local variable, then use the `resultVector` to scan other 2048 offsets in storage. Then we can scan data from storage based on those offsets. Finally, we populate list entries in `resultVector`. This can avoid dynamic allocations, but complicate the whole logic a lot.